guides/generate_punks_with_gan.md

```diff
@@ -0,0 +1,164 @@
+            nn.ReLU(True),
+            nn.ConvTranspose2d(ngf, nc, 4, 2, 1, bias=False),
+            nn.Tanh(),
+        )
+
+    def forward(self, input):
+        output = self.network(input)
+        return output
+```
+
+We're taking the generator from [this repo by @teddykoker](https://github.com/teddykoker/cryptopunks-gan/blob/main/train.py#L90), where you can also see the original discriminator model structure.
+
+After instantiating the model, we'll load in the weights from the Hugging Face Hub, stored at [nateraw/cryptopunks-gan](https://huggingface.co/nateraw/cryptopunks-gan):
+
+```python
+from huggingface_hub import hf_hub_download
+import torch
+
+model = Generator()
+weights_path = hf_hub_download('nateraw/cryptopunks-gan', 'generator.pth')
+model.load_state_dict(torch.load(weights_path, map_location=torch.device('cpu'))) # Use 'cuda' if you have a GPU available
+```
+
+## Step 2 — Defining a `predict` function
+
+The `predict` function is the key to making Gradio work! Whatever inputs we choose through the Gradio interface will get passed through our `predict` function, which should operate on the inputs and generate outputs that we can display with Gradio output components. For GANs it's common to pass random noise in to our model as the input, so we'll generate a tensor of random numbers and pass that through the model. We can then use `torchvision`'s `save_image` function to save the output of the model as a `png` file, and return the file name:
+
+```python
+from torchvision.utils import save_image
+
+def predict(seed):
+    num_punks = 4
+    torch.manual_seed(seed)
+    z = torch.randn(num_punks, 100, 1, 1)
+    punks = model(z)
+    save_image(punks, "punks.png", normalize=True)
+    return 'punks.png'
+```
+
+We're giving our `predict` function a `seed` parameter, so that we can fix the random tensor generation with a seed. We'll then be able to reproduce punks if we want to see them again by passing in the same seed.
+
+*Note!* Our model needs an input tensor of dimensions 100x1x1 for to do a single inference, or (BatchSize)x100x1x1 for generating a batch of images. In this demo we'll start by generating 4 punks at a time.
+
+## Step 3 — Creating a Gradio interface
+
+At this point you can even run the code you have with `predict(<SOME_NUMBER>)`, and you'll find your freshly generated punks in your file system at `./punks.png`. To make a truly interactive demo, though, we'll build out a simple interface with Gradio. Our goals here are to:
+
+* Set a slider input so users can choose the "seed" value
+* Use an image component for our output to showcase the generated punks
+* Use our `predict()` to take the seed and generate the images
+
+With `gr.Interface()`, we can define all of that with a single function call:
+
+```python
+import gradio as gr
+
+gr.Interface(
+    predict,
+    inputs=[
+        gr.inputs.Slider(label='Seed', minimum=0, maximum=1000, default=42),
+    ],
+    outputs="image",
+).launch()
+```
+
```

Comment:
Suggestion: this would be a good place to embed an interface so readers can immediately interact with what's been built so far (before the second slider). To do so you'll need to create a version of the space from this gradio config